src/Product/Search.php

```diff
@@ -132,6 +134,30 @@ public function initSearch($selectedFilters)
         $this->getSearchAdapter()->useFiltersAsInitialPopulation();
     }
 
+    public function getProductIdsUsingCoreSearch()
+    {
+        // Search using the core functionality
+        $result = SearchCore::find(
+            $this->context->language->id,
+            Tools::replaceAccentedChars(urldecode($this->query->getSearchString())),
+            1,
+            PHP_INT_MAX,
+            'position',
+            'desc',
+            false,
+            false,
+            null
+        );
+
+        // Extract IDs, return some nonsense
```

Comment:
explain more in this comment, or remove it.
Now it just raises questions why are you returning "nonsense"